Neos.ContentGraph.DoctrineDbalAdapter/src/ContentGraphFactory.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Neos\ContentGraph\DoctrineDbalAdapter;
+
+use Neos\ContentGraph\DoctrineDbalAdapter\Domain\Repository\ContentGraph;
+use Neos\ContentGraph\DoctrineDbalAdapter\Domain\Repository\NodeFactory;
+use Neos\ContentRepository\Core\ContentGraphFactoryInterface;
+use Neos\ContentRepository\Core\Infrastructure\DbalClientInterface;
+use Neos\ContentRepository\Core\NodeType\NodeTypeManager;
+use Neos\ContentRepository\Core\Projection\Workspace\Workspace;
+use Neos\ContentRepository\Core\Projection\Workspace\WorkspaceStatus;
+use Neos\ContentRepository\Core\SharedModel\ContentRepository\ContentRepositoryId;
+use Neos\ContentRepository\Core\SharedModel\Exception\ContentStreamDoesNotExistYet;
+use Neos\ContentRepository\Core\SharedModel\Workspace\ContentStreamId;
+use Neos\ContentRepository\Core\SharedModel\Workspace\WorkspaceDescription;
+use Neos\ContentRepository\Core\SharedModel\Workspace\WorkspaceName;
+use Neos\ContentRepository\Core\SharedModel\Workspace\WorkspaceTitle;
+
+/**
+ * @internal only used inside the
+ * @see ContentGraphFinder
+ */
+readonly final class ContentGraphFactory implements ContentGraphFactoryInterface
+{
+    public function __construct(
+        private DbalClientInterface $client,
+        private NodeFactory $nodeFactory,
+        private ContentRepositoryId $contentRepositoryId,
+        private NodeTypeManager $nodeTypeManager,
+        private string $tableNamePrefix
+    ) {
+    }
+
+    public function buildForWorkspace(WorkspaceName $workspaceName): ContentGraph
+    {
+        // FIXME: Should be part of this projection, this is forbidden
+        $tableName = strtolower(sprintf(
+            'cr_%s_p_%s',
+            $this->contentRepositoryId->value,
+            'Workspace'
+        ));
+
+        $row = $this->client->getConnection()->executeQuery(
+            '
+                SELECT * FROM ' . $tableName . '
+                WHERE workspaceName = :workspaceName
```

Comment:
Not sure if that's an improvement, so feel free to ignore
```suggestion
                WHERE workspaceName = :workspaceName
                LIMIT 1
```